src/libs/migrations/KeyReportActionsDraftByReportActionID.js

```diff
@@ -0,0 +1,57 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * This migration updates reportActionsDrafts data to be keyed by reportActionID.
+ *
+ * Before: reportActionsDrafts_reportID_reportActionID: value
+ * After: reportActionsDrafts_reportID: {[reportActionID]: value}
+ *
+ * @returns {Promise}
+ */
+export default function () {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS,
+            waitForCollectionCallback: true,
+            callback: (allReportActionsDrafts) => {
+                Onyx.disconnect(connectionID);
+
+                if (!allReportActionsDrafts) {
+                    Log.info('[Migrate Onyx] Skipped migration KeyReportActionsDraftByReportActionID because there were no reportActionsDrafts');
+                    return resolve();
+                }
+
+                const newReportActionsDrafts = {};
+                _.each(allReportActionsDrafts, (reportActionDraft, onyxKey) => {
+                    if (!_.isString(reportActionDraft)) {
+                        return;
+                    }
+                    newReportActionsDrafts[onyxKey] = null;
+
+                    if (_.isEmpty(reportActionDraft)) {
+                        return;
+                    }
+
+                    const reportActionID = onyxKey.split('_').pop();
+                    const newOnyxKey = onyxKey.replace(`_${reportActionID}`, '');
+                    newReportActionsDrafts[newOnyxKey] = {
+                        ...(newReportActionsDrafts[newOnyxKey] || allReportActionsDrafts[newOnyxKey]),
```

Comment:
What does `|| allReportActionsDrafts[newOnyxKey]` do in here? do we ever set `allReportActionsDrafts[newOnyxKey]`?

Reply:
Can you add a comment to make that obvious please? I see the test, and with one of the values for reportID 2 set to `{}` and one to null, but is that something that can happen, like having a migration run half-way?